### PR TITLE
Move publish to github action 

### PR DESCRIPTION
**Pull request by pblivin0x**

### CODE REVIEW

File: .github/publish.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        run: cp .env.default .env
+      - name: install
+        run: yarn
+      - name: prepublish
+        run: yarn prepublishOnly
+      - name: release
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+        run: npx semantic-release --branches master
```
Comment (FlattestWhite) [inline .github/publish.yml:48]: I think we should keep the yarn publish:ci:latest

Comment (pblivin0x) [reply .github/publish.yml:48]: was referencing `index-coop-smart-contracts` https://github.com/IndexCoop/index-coop-smart-contracts/blob/add917bec120cce4cf0fd63d0136fbc72aad16fa/.github/workflows/publish.yaml#L48

Comment (pblivin0x) [reply .github/publish.yml]: added here [9dd3b79](https://github.com/IndexCoop/index-protocol/pull/13/commits/9dd3b79021089e373cce190bbdfef2f906ff6249)

**Review by FlattestWhite — APPROVED** [review]
LGTM

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - name: build
+        run: yarn build
+      - name: test
+        run: |
+          yarn chain &
+          yarn test
+
+  publish:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/master' }}
```
Comment (FlattestWhite) [inline .github/workflows/publish.yml:31]: publish will only run if its master btw

**Comment by FlattestWhite** [comment]
Just fixed the credentials with GH_TOKEN

**Pull request closed without merging**